lib/pavilion/series.py

```diff
@@ -36,12 +43,173 @@ def test_obj_from_id(pav_cfg, test_ids):
+
+            for next in next_str_list:
+                self.test_sets[set_name].add_next(self.test_sets[next])
+
+        # CLEAN UP ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+        self.print_stats()
+
+        # kick off tests that aren't waiting on anyone
+        self.currently_running = []
+        for set_name in self.test_sets:
+            if not self.test_sets[set_name].get_prev():
+                self.test_sets[set_name].run_set()
+                self.currently_running.append(self.test_sets[set_name])
+
+        self.print_stats()
+
+        # keep checking on statuses of currently running sets
+        for set_obj in self.currently_running:
+            time.sleep(5)
+            if self.is_set_done(set_obj):
+                set_obj.change_stat('DONE')
+                for waiting in set_obj.get_next():
+                    run_this = True
+                    for prev in waiting.get_prev():
+                        if prev.get_stat() != 'DONE':
+                            run_this = False
+                    if run_this:
+                        waiting.run_set()
+                        self.currently_running.append(waiting)
+
+            self.print_stats()
+
+
+        # END CLEAN UP
+
+
+
+    def is_set_done(self, set_obj):
+        for test in set_obj.test_runs:
+            if not pathlib.Path(test.path/'RUN_COMPLETE').exists():
+                return False
+        return True
+
+
+    def make_dep_graph(self):
+        # has to be a graph of test sets
+        for set_name in self.sets:
+            self.dep_graph[set_name] = self.sets[set_name]['depends_on']
+
+        dbg_print(self.dep_graph, '\n')
+
+    def print_stats(self):
+        # dbg_print stats of sets
+        for set_name in self.test_sets:
+            ts = self.test_sets[set_name]
+            dbg_print(ts, ': ', ts.get_stat(), ts.test_runs)
+
+        dbg_print()
+
+
+class TestSet:
+
+    # statuses:
+    # NO_STAT, NEXT, DID_NOT_RUN, RUNNING, PASS, FAIL
+
+    def __init__(self, _pav_cfg, _name, _args_list,
+                 _prev_set, _next_set):
+
+        self.name = _name
+        self.pav_cfg = _pav_cfg
+        self.args_list = _args_list
+        self.prev_set = _prev_set  # has to be a list of TestSet objects
+        self.next_set = _next_set  # has to be a list of TestSet objects
+        self.status = 'NO_STAT'
+        self.test_runs = list()
+
+    def run_set(self):
+        run_cmd = commands.get_command('run')
+        arg_parser = arguments.get_parser()
+        args = arg_parser.parse_args(self.args_list)
+        run_cmd.run(self.pav_cfg, args)
+        self.test_runs.extend(run_cmd.last_tests)
+
+        # change statuses
+        self.change_stat('RUNNING')
+        for n_set in self.next_set:
+            if n_set.get_stat() == 'NO_STAT':
+                n_set.change_stat('NEXT')
+
+    def change_stat(self, new_stat):
+        self.status = new_stat
+
+    def get_stat(self):
```

Comment:
I really don't like your abbreviation of 'status' to 'stat'. Stat has another well defined meaning (ie os.stat), and it's only two more characters.  Plus if you're tracking status, why not just reuse existing status file code?